src/main/java/de/qabel/core/drop/DropMessage.java

```
@@ -1,80 +1,102 @@
+     */
+    DropMessage(String senderKeyId, T data, Date created, String acknowledgeId) {
+		this.senderKeyId = senderKeyId;
+		this.data = data;
+		this.created = created;
+		this.acknowledgeId = acknowledgeId;
+	}
+
+	public static int getVersion() {
+        return VERSION;
     }
 
-    public void setTime(Date time) {
-        this.time = time.getTime();
+    public Date getCreationDate() {
+        return created;
     }
 
     public String getAcknowledgeID() {
-        return acknowledgeID;
+        return acknowledgeId;
     }
 
-    public void setAcknowledgeID(String acknowledgeID) {
-        this.acknowledgeID = acknowledgeID;
-    }
-
-    public String getSender() {
+    public Entity getSender() {
         return sender;
     }
 
-    public void setSender(String sender) {
-        this.sender = sender;
+    String getSenderKeyId() {
+    	return senderKeyId;
     }
 
-    public Class<T> getModelObject() {
-        return modelObject;
+    /**
+     * Register the given Entity as sender of this drop message
+     * and check if it matches the senderKeyId.
+     * This is used to complete the deserialization of DropMessage.
+     *
+     * @param sender Entity to be registered as sender.
+     * @return true if the given sender matches the senderKeyId, otherwise false.
+     */
+    boolean registerSender(Entity sender) {
+    	if (!senderKeyId.equals(sender.getKeyIdentifier())) {
+    		return false;
+    	}
+    	this.sender = sender;
+    	return true;
     }
 
-    public void setModelObject(Class<T> modelObject) {
-        this.modelObject = modelObject;
+    public Class<T> getModelObject() {
+        return (Class<T>)data.getClass();
     }
 
     public T getData() {
         return data;
     }
 
-    public void setData(T data) {
-        this.data = data;
+    /**
+     * Enable/disable acknowledge request for this drop message.
+     *
+     * @param enabled
+     */
+    public void enableAcknowledgeing(boolean enabled) {
```

I'll prevent re-generation of ackIds if they are already set. But if NOACK will be be the new default, this function should no longer be misleading, so I can keep the rest as it is.